Firmware/planner.cpp

```diff
@@ -459,10 +459,7 @@ void plan_init() {
   #ifdef LIN_ADVANCE
   memset(position_float, 0, sizeof(position_float)); // clear position
   #endif
-  previous_speed[0] = 0.0;
-  previous_speed[1] = 0.0;
-  previous_speed[2] = 0.0;
-  previous_speed[3] = 0.0;
+  memset(previous_speed, 0, sizeof(previous_speed));
```

Comment:
I'm surprised the compiler couldn't optimize the float assignments into a single block of "memset"

Reply:
Before:
```
  previous_speed[0] = 0.0;
   3152a:	sts	0x0532, r1	; 0x800532 <_ZL14previous_speed.lto_priv.327>
   3152e:	sts	0x0533, r1	; 0x800533 <_ZL14previous_speed.lto_priv.327+0x1>
   31532:	sts	0x0534, r1	; 0x800534 <_ZL14previous_speed.lto_priv.327+0x2>
   31536:	sts	0x0535, r1	; 0x800535 <_ZL14previous_speed.lto_priv.327+0x3>
  previous_speed[1] = 0.0;
   3153a:	sts	0x0536, r1	; 0x800536 <_ZL14previous_speed.lto_priv.327+0x4>
   3153e:	sts	0x0537, r1	; 0x800537 <_ZL14previous_speed.lto_priv.327+0x5>
   31542:	sts	0x0538, r1	; 0x800538 <_ZL14previous_speed.lto_priv.327+0x6>
   31546:	sts	0x0539, r1	; 0x800539 <_ZL14previous_speed.lto_priv.327+0x7>
  previous_speed[2] = 0.0;
   3154a:	sts	0x053A, r1	; 0x80053a <_ZL14previous_speed.lto_priv.327+0x8>
   3154e:	sts	0x053B, r1	; 0x80053b <_ZL14previous_speed.lto_priv.327+0x9>
   31552:	sts	0x053C, r1	; 0x80053c <_ZL14previous_speed.lto_priv.327+0xa>
   31556:	sts	0x053D, r1	; 0x80053d <_ZL14previous_speed.lto_priv.327+0xb>
  previous_speed[3] = 0.0;
   3155a:	sts	0x053E, r1	; 0x80053e <_ZL14previous_speed.lto_priv.327+0xc>
   3155e:	sts	0x053F, r1	; 0x80053f <_ZL14previous_speed.lto_priv.327+0xd>
   31562:	sts	0x0540, r1	; 0x800540 <_ZL14previous_speed.lto_priv.327+0xe>
   31566:	sts	0x0541, r1	; 0x800541 <_ZL14previous_speed.lto_priv.327+0xf>
```
After:
```
  memset(previous_speed, 0, sizeof(previous_speed));
   3152c:	ldi	r30, 0x32	; 50
   3152e:	ldi	r31, 0x05	; 5
   31530:	movw	r26, r30
   31532:	st	X+, r1
   31534:	dec	r24
   31536:	brne	.-6      	; 0x31532 <main+0xae0>
```